### PR TITLE
Notification settings

### LINKED ISSUE #2346: Allow enable / disabling of news message notification types
I looked through the issue history and couldn't find one on this:

I can't remember the last time I saw a notification that a ride had broken down and thought to myself, "wow! That's useful information! I will do something about that!" No - a ride breaking down is not noteworthy.

Furthermore, in anything other than a small park, the notification queue just gets spammed with these. A new ride was unlocked? I want to know about that. A ride "still hasn't been fixed?" I want to know about that. Guests are complaining about the disgusting state of the paths in my park? I want to know about that. Etc.

The problem is that when the notifications are so rarely notworthy, I don't even notice the messages I care about, because I'm getting so many breakdown notifications. (In my current park one every 10 seconds or so.) If I open the notification history, I have only one non-breakdown notification, because they happen so much more often than anything else.

Breakdown notifications aren't useful information. Is it possible to add a way to turn them off?

### PR DESCRIPTION
**Pull request by IntelOrca**
Since its Christmas, I thought I'd implement a significant feature. So here is the window that allows you to enable and disable different types of notifications. No more breakdown messages!

Just open up the recent messages window and click the settings icon to access the notification settings window.

Guest is lost (and can't find the park exit) notification is the really annoying notification from the first game that was removed in RCT2. I haven't added the actual notification yet (but the option is here for when it is) - I thought it would be funny to allow it to be enabled for nostalgia I guess.

Note I have grouped some notifications together, like guest warnings. I don't think it will be common for a player to enable e.g. warning for hungry guests and not the warning for guests needing the toilet.

Closes #2346.

Feedback welcome. :)

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Shouldn't we modify news item queue rather than have all the extra if statements that might be missed in the calling functions?

**Comment by IntelOrca** [comment]
@duncanspumpkin Well we can't do that unless we add an enum for every type of notification.

**Comment by janisozaur** [comment]
Don't forget to update changelog.

**Comment by IntelOrca** [comment]
@janisozaur :+1:

File: data/language/english_uk.txt [hunk 1]
```diff
@@ -3924,6 +3924,24 @@ STR_5582    :Trap mouse cursor in window
 STR_5583    :{COMMA1DP16}ms{POWERNEGATIVEONE}
 STR_5584    :SI
 STR_5585    :{SMALLFONT}{BLACK}Unlocks ride operation limits, allowing for things like {VELOCITY} lift hills
+STR_5589    :Notification Settings
+STR_5590    :Park awards
+STR_5591    :Marketing campaign has finished
+STR_5592    :Park warnings
+STR_5593    :Park rating warnings
+STR_5594    :Ride has broken_down
```
Comment (mrtnptrs) [inline data/language/english_uk.txt]: Remove the underscore (_) here :)

**Comment by rd3k** [comment]
Excellent feature - thanks Ted!

**Comment by AaronVanGeffen** [comment]
Love the idea. From a user's perspective, though, is there a technical reason why it wasn't implemented as a tab in the options window? I think it would be more easy to discover there, however, I recall something about the number of widgets per tab being limited? That would hinder putting it all in one tab, I guess...

**Comment by IntelOrca** [comment]
@AaronVanGeffen Because there are 3 tabs for this window alone. Themes also have their own window.

**Pull request merged**